src/pages/Protect/Protect.jsx

```diff
@@ -1,286 +1,62 @@
-              {
-                  "name": "",
-                  "type": "uint256"
-              }
-          ],
-          "payable": false,
-          "stateMutability": "view",
-          "type": "function"
-      },
-      {
-          "payable": true,
-          "stateMutability": "payable",
-          "type": "fallback"
-      },
-      {
-          "anonymous": false,
-          "inputs": [
-              {
-                  "indexed": true,
-                  "name": "owner",
-                  "type": "address"
-              },
-              {
-                  "indexed": true,
-                  "name": "spender",
-                  "type": "address"
-              },
-              {
-                  "indexed": false,
-                  "name": "value",
-                  "type": "uint256"
-              }
-          ],
-          "name": "Approval",
-          "type": "event"
-      },
-      {
-          "anonymous": false,
-          "inputs": [
-              {
-                  "indexed": true,
-                  "name": "from",
-                  "type": "address"
-              },
-              {
-                  "indexed": true,
-                  "name": "to",
-                  "type": "address"
-              },
-              {
-                  "indexed": false,
-                  "name": "value",
-                  "type": "uint256"
-              }
-          ],
-          "name": "Transfer",
-          "type": "event"
-      }
-    ];
-   
-    const contract = await new web3.eth.Contract(abi2,_token)
+  const approve_weth = async (_value, _token) => {
+    const contract = await new web3.eth.Contract(ERC20_ABI, _token);
     var meth = contract.methods;
-    
-    if(address!=null){
-     console.log(meth)
-  let event =     await meth.approve('0x04796D80B66544EF9C4A08A5477E35C1632719f9', web3.utils.toBN(_value*1e18)).send({from:address,value: 0}).then(console.log);
-     
-    } else {
-      console.log('Wallet not connected!')
-    }
-  }
-  let disable = true;
-  const ls = async(e)=>{
-    console.log("pp")
-    // e.preventDefault()
-    // approve_weth(0.001, '0xd0A1E359811322d97991E03f863a0C30C2cF029C')
-    approve_weth(protectedAmount, '0xd0A1E359811322d97991E03f863a0C30C2cF029C', address)
-
-  }
 
-  const input_fill = ()=>{
-    window.alert("please fill the proper information before Hedge  WETH ")
-  }
 
-  setTotalLimit( price*protectedAmount ) // karan will help us 
+    if (address != null) {
+      console.log(meth);
+      let event = await meth
+        .approve(
+          "0x72D28BCa958f45aEC793df2E62a1b19a9C4c4d4d",
+          web3.utils.toBN(_value * 1e18)
+        )
+        .send({ from: address, value: 0 })
+        .then(console.log);
 
+    } else {
+      console.log("Wallet not connected!");
+    }
+  };
+  let disable = true;
+  const approve_Weth = async (e) => {
+    approve_weth(
+      protectedAmount,
+      "0xd0A1E359811322d97991E03f863a0C30C2cF029C",
```

Comment:
"0xd0A1E359811322d97991E03f863a0C30C2cF029C"
can we pick this from ASSET_VALUES instead of giving this stray address here ?
Whatever asset this address belongs to, add it as key and the address as value and pick it up here.

Reply:
hat this is already done in the miner changes branch
just let me know when you are free so that we can work with th

Reply:
sure